### ISSUE TITLE
[New NIP]: Introducing: NeblioX, Rebuilding Neblio Technology Using Latest Bitcoin Core

### ISSUE THREAD

**Issue by NeblioTeam** [NeblioTeam/Neblio-Improvement-Proposals#8]
### Proposal Description

This proposal was written by the Neblio Core Development Team and is being presented to the community to gather feedback and ideas.

## Introduction

Neblio was forked from Bitcoin Core a very long time ago. It, hence, didn't receive the updates Bitcoin received over time. While this is not necessarily a bad thing, Neblio can benefit from the high volume of updates bitcoin core is receiving, while building over its technology, such as SegWit, Lightning, Taproot, and more. 

## Benefits and Future Prospects

Neblio's future plans include easy and cheap transfers of digital assets and information over multiple networks. We've learned from Ethereum and other congested blockchains that doing this all on-chain is not scalable, leading to extremely high transaction costs, reaching over $100 per transaction, and leading also to centralization. Ethereum nodes cannot be maintained by normal people, but by coordinated leadership, due to the gigantic blockchain size (currently 1 TB). Realistically speaking, if one buys gum, that transaction shouldn't be stored on the blockchain, as this eventually defeats the idea that everyone can run a node and everyone can be part of the network, as opposed to being just a user of the network. 

Despite Neblio having a current capacity (transactions-per-second, block size capacity, etc) that far-exceeds other blockchains, we must always be preparing for and developing for the future. 

Hence, a good future for decentralized cryptocurrency will include a second layer, such as the Lightning Network. With Bitcoin's Taproot activation, lightning channels will become more efficient and cheaper. Neblio can benefit from this by using the taproot tools to transfer assets off-chain, allowing users to benefit from the fast and cheap transactions with instant settlement. 

**Instant settlement over Lighting will also allow advanced Decentralized Exchanges (DeX's), instant P2P trading, along with new Real-time Apps, to be built on top of Neblio, such as messaging apps, instant tipping, streaming funds and/or assets, and more.**

Another benefit of this is that it'll make it much easier to integrate with hardware wallets such as Ledger and Trezor. Bitcoin Core currently supports HD wallets, while also supporting a wallet system where multiple wallets can be opened and used simultaneously, which will make adding hardware wallets much easier. Moreover, Bitcoin is moving soon to sqlite (instead of Berkeley DB) as its wallet storage. This will solve some of Neblio's problems for big wallets, which bitcoin itself has been facing, in addition to many other problems introduced by the inefficient and outdated Berkeley DB.

Many other features will be gained, to name a few, replacement by fee (RBF) that's optional and decided by the user, Schnorr signatures, SegWit to avoid transaction malleability, more privacy with Taproot, and most of the BIPs Bitcoin implemented over the years that Neblio is lacking.

## Execution Plan

The question is whether it's worth it to proceed with this project. The only way to know is to start for a short period of time and assess the technical risks and the possible issues that will come with this project. An initial plan will be to start the development starting from the current Bitcoin Core until the end of January, and by then reassess the cost/benefit of the project.

### Proposal Voting Details

We propose that the vote for this NIP lasts for 86,400 blocks or roughly 30 days. 

### Other Information

None

**Comment by NeblioTeam** [comment]
https://github.com/NeblioTeam/NeblioX Has been created for Research & Development in this area.

**Comment by Shane-Neblio** [comment]
only if it's needed & as long as it doesn't screw up whatever Tomas has done. you don't want those factories running into a problem. 

why call it NeblioX? new token listing? rebranding?

**Comment by NeblioTeam** [comment]
`NeblioX` is simply the project name to differentiate it from the current Neblio Core.

**Comment by RasmonT** [comment]
It's like it will be possible to bridge NeblioX asset also to the Neblio mainnet and vice versa? For example I will send neblios through the NeblioX to the Neblio mainnet? That would be great to have second layer with this option. Let me know further! As I'm excited!

**Comment by NeblioTeam** [comment]
NeblioX is anticipated to be fully compatible with the Neblio Mainnet. It is an upgrade/rewrite of Neblio Core on top of a stronger foundation which will provide more features and better performance. It should, eventually, have all features of Neblio Core, plus new ones.

**Comment by RasmonT** [comment]
> NeblioX is anticipated to be fully compatible with the Neblio Mainnet. It is an upgrade/rewrite of Neblio Core on top of a stronger foundation which will provide more features and better performance. It should, eventually, have all features of Neblio Core, plus new ones.

Glad to hear that! Sounds very interesting.

**Comment by RasmonT** [comment]
@NeblioTeam Will the NeblioX supports possibilities of ZK rollups? It's seems like NeblioX will be ZK rollup for Neblio network, but few chances are trying to be ZK rollup for ETH network, is there any chance NeblioX will be also aiming this field or not?

**Comment by NeblioTeam** [comment]
I assume you mean ZK rollups as they are often referred to in Ethereum circles. If so, no, NeblioX will not need or use anything like that. NeblioX will get all of the benefits and features added to Bitcoin Core over the last ~5 years that Neblio lacks, such as SegWit, Lightning, Taproot, and MANY other BIPs. 

These solutions, and the UTXO model, are far superior to ZK rollups and the account model used in Ethereum and its clones, imho.

**Comment by NeblioTeam** [comment]
Unless there is opposition or discussion around why this proposal should not be moved forward, we plan to finalize this proposal, lock it in, and schedule it for a vote by Dec 10th (one week after it was originally published here).

**Comment by NeblioTeam** [comment]
This proposal is officially `NIP8` and is finalized and ready for scheduling. Any inputs from anyone here on when this should be scheduled? After the holidays may be best.

**Comment by RyanXNeb** [comment]
> This proposal is officially `NIP8` and is finalized and ready for scheduling. Any inputs from anyone here on when this should be scheduled? After the holidays may be best.

Yes I also think after the holidays to get the max amount of participation in voting.

**Comment by Oepsje** [comment]
It'd say let the voting start as soon as possible. Some people have more time to check the proposal / vote during the holidays, while others have more time during normal weeks. I think this would result in the max amount of participation. Furthermore the sooner we have the results the sooner we can get the ball roling

**Comment by NeblioTeam** [comment]
`NIP8` has been scheduled for voting!

Vote Start: Block 3647300 (Approximately Jan 5th 2022)
Vote End: Block 3733700 (Approximately Feb 4th 2022)

If you support `NIP8`, set your Neblio Staking wallets (Neblio Core v3.4.0 or higher required) to vote `1` for proposal ID `8` starting at block `3647300` and ending at block `3733700` now!

**Comment by NeblioTeam** [comment]
NIP8 has officially been approved by NEBL stakers! https://explorer.nebl.io/voting

Development work on NIP8 is already underway: https://github.com/NeblioTeam/NeblioX

You can track NIP8's progress on the [Neblio Roadmap](https://github.com/NeblioTeam/Neblio-Improvement-Proposals/projects/1).
